legal-api/tests/unit/services/test_authorization.py

```diff
@@ -80,6 +92,144 @@ def get():
 ]
 
 
+class FilingKey(str, Enum):
+    """Define an enum for expected allowable filing keys."""
+
+    ADMN_FRZE = 'ADMN_FRZE'
+    IA_CP = 'IA_CP'
+    IA_BC = 'IA_BC'
+    IA_BEN = 'IA_BEN'
+    IA_CC = 'IA_CC'
+    IA_ULC = 'IA_ULC'
+    REG_SP = 'REG_SP'
+    REG_GP = 'REG_GP'
+    CORRCTN = 'CORRCTN'
+    CORRCTN_FIRMS = 'CORRCTN_FIRMS'
+    AR_CP = 'AR_CP'
+    AR_CORPS = 'AR_CORPS'
+    COA_CP = 'COA_CP'
+    COA_CORPS = 'COA_CORPS'
+    COD_CP = 'COD_CP'
+    COD_CORPS = 'COD_CORPS'
+    COURT_ORDER = 'COURT_ORDER'
+    VOL_DISS = 'VOL_DISS'
+    VOL_DISS_FIRMS = 'VOL_DISS_FIRMS'
+    REGISTRARS_NOTATION = 'REGISTRARS_NOTATION'
+    REGISTRARS_ORDER = 'REGISTRARS_ORDER'
+    SPECIAL_RESOLUTION = 'SPECIAL_RESOLUTION'
+    ALTERATION = 'ALTERATION'
+    CONTINUATION_OUT = 'CONTINUATION_OUT'
+    TRANSITION = 'TRANSITION'
+    CHANGE_OF_REGISTRATION = 'CHANGE_OF_REGISTRATION'
+    CONV_FIRMS = 'CONV_FIRM'
+    RESTRN_FULL_CP = 'RESTRN_FULL_CP'
+    RESTRN_FULL_CORPS = 'RESTRN_FULL_CORPS'
+    RESTRN_LTD_CORPS = 'RESTRN_LTD_CORPS'
+    RESTRN_LTD_EXT_CORPS = 'RESTRN_LTD_EXT_CORPS'
+    RESTRN_LTD_EXT_LLC = 'RESTRN_LTD_EXT_LLC'
+    RESTRN_LTD_TO_FULL_CORPS = 'RESTRN_LTD_TO_FULL_CORPS'
+    RESTRN_LTD_TO_FULL_LLC = 'RESTRN_LTD_TO_FULL_LLC'
+    PUT_BACK_ON = 'PUT_BACK_ON'
+
+
+EXPECTED_DATA = {
+    FilingKey.ADMN_FRZE: {'displayName': 'Admin Freeze', 'feeCode': 'NOFEE', 'name': 'adminFreeze'},
+    FilingKey.AR_CP: {'displayName': 'Annual Report', 'feeCode': 'OTANN', 'name': 'annualReport'},
+    FilingKey.AR_CORPS: {'displayName': 'Annual Report', 'feeCode': 'BCANN', 'name': 'annualReport'},
+    FilingKey.COA_CP: {'displayName': 'Address Change', 'feeCode': 'OTADD', 'name': 'changeOfAddress'},
+    FilingKey.COA_CORPS: {'displayName': 'Address Change', 'feeCode': 'BCADD', 'name': 'changeOfAddress'},
+    FilingKey.COD_CP: {'displayName': 'Director Change', 'feeCode': 'OTCDR', 'name': 'changeOfDirectors'},
+    FilingKey.COD_CORPS: {'displayName': 'Director Change', 'feeCode': 'BCCDR', 'name': 'changeOfDirectors'},
```

Comment:
It's nice that the fee code is passed from the BE to the UI -- so the UI shouldn't need to have its own copy of the fee codes enum -- but what about the "free" filings? Ie,
BCFDR - free director change BC company
OTFDR - free director change OT(her)

^^ The free ARs and COAs are handled without a special code, I think.

Reply:
I wonder if we could change the free COD mechanism to be like the free ARs/COAs? (I think it's done with a flag instead of a special code.)

If you agree then I'll create a new ticket.